source/vibe/web/rest.d

```diff
@@ -24,47 +240,77 @@ import std.typetuple : anySatisfy, Filter;
-	POST is used.
-
-	The following table lists the mappings from prefix verb to HTTP verb:
-
-	$(TABLE
-		$(TR $(TH Prefix) $(TH HTTP verb))
-		$(TR $(TD get)	  $(TD GET))
-		$(TR $(TD query)  $(TD GET))
-		$(TR $(TD set)    $(TD PUT))
-		$(TR $(TD put)    $(TD PUT))
-		$(TR $(TD update) $(TD PATCH))
-		$(TR $(TD patch)  $(TD PATCH))
-		$(TR $(TD add)    $(TD POST))
-		$(TR $(TD create) $(TD POST))
-		$(TR $(TD post)   $(TD POST))
-	)
-
-	If a method has its first parameter named 'id', it will be mapped to ':id/method' and
-	'id' is expected to be part of the URL instead of a JSON request. Parameters with default
-	values will be optional in the corresponding JSON request.
-
-	Any interface that you return from a getter will be made available with the
-	base url and its name appended.
-
-	Params:
-		router = The HTTP router on which the interface will be registered
-		instance = Class instance to use for the REST mapping - Note that TImpl
-			must either be an interface type, or a class which derives from a
-			single interface
-		settings = Additional settings, such as the $(D MethodStyle), or the prefix.
-			See $(D RestInterfaceSettings) for more details.
-
-	See_Also:
-		$(D RestInterfaceClient) class for a seamless way to access such a generated API
-
-*/
+ * Registers a server matching a REST interface.
+ *
+ * Servers are implementation of an interface. Consequently, they are classes
+ * that implement an interface that obey the rule defined in this module's
+ * documentation.
+ *
+ * A basic 'hello world' API can be defined as follow:
+ * ----
+ * @path("/api/")
+ * interface APIRoot {
+ *     string get();
+ * }
+ *
+ * class API : APIRoot {
+ *     override string get() { return "Hello, World"; }
+ * }
+ *
+ * shared static this() {
+ *     auto settings = new HTTPServerSettings;
+ *     settings.port = 8080;
+ *     settings.bindAddresses = ["::1", "127.0.0.1"];
+ *     auto router = new URLRouter;
+ *
+ *     // -- Where the magic happens --
+ *     router.registerRestInterface(new API());
+ *     // Now you just need to GET /api/ and you'll get 'Hello, World'.
+ *     listenHTTP(settings, router);
+ * }
+ * ----
+ *
+ * As you can see, REST's logic can be written inside the object without any
+ * concern for HTTP matters.
+ *
+ * Return_value:
+ * By default, all method that return a value send a 200 code, or 204 if no
+ * value is being returned for the body.
+ *
+ * Non-success:
+ * In the case where you want to return an error code, you can throw an
+ * `HTTPStatusException` from within your business code. It will be turned into
+ * a Json object that'll have a `statusMessage` with the exception message.
+ * In the case of other exception being thrown, a Json object containing a
+ * `statusMessage` set to the exception message and a `statusDebugMessage`
+ * set to the complete string representation of the `Exception` (== `toString`)
+ * will be returned, with an error code of 500.
```

Comment:
I really think we should also come up with a solution to avoid throwing exceptions for the return value.